cmake/CompilerFlags.cmake

```diff
@@ -0,0 +1,453 @@
+          		set(OPT_FFLAGS ${OPT_FFLAGS} -mtune=native)
+        	else() # i386 needs to be told to use sse prior to using -mfpmath=sse
+          		set(OPT_FFLAGS ${OPT_FFLAGS} -mtune=native -msse -mfpmath=sse)
+         	endif()
+         endif()
+	endif()
+	
+	
+	# gcc 4.1.2 does not support putting allocatable arrays in a Fortran type...
+    # so unfortunately file-less prmtop support in the sander API will not work
+    # in this case.
+    if("${CMAKE_Fortran_COMPILER_VERSION}" VERSION_LESS 4.2)
+        add_definitions(-DNO_ALLOCATABLES_IN_TYPE)
+    endif()
+    
+    # Check dragonegg
+	if(DRAGONEGG)
+		#TODO: write check_fortran_compiler_flag
+		#check_fortran_compiler_flag(-fplugin=${DRAGONEGG} DRAGONEGG_FORTRAN_WORKS)
+		#if(NOT DRAGONEGG_FORTRAN_WORKS)
+		#	message(FATAL_ERROR "Can't use Fortran compiler with Dragonegg.  Please fix whatever's broken.  Check CMakeOutput.log for details.")
+		#endif()
+		
+		add_flags(Fortran -fplugin=${DRAGONEGG})
+	endif()
+endif()
+
+#clang
+#---------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "AppleClang")
+	add_flags(C -Wall -Wno-unused-function)
+	
+	list(APPEND OPT_CFLAGS "-mtune=native")
+	
+	#if we are crosscompiling and using clang, tell CMake this
+	if(CROSSCOMPILE)
+		set(CMAKE_C_COMPILER_TARGET ${TARGET_TRIPLE})
+	endif()  
+	
+	if(NOT UNINITIALIZED_WARNINGS)
+		add_flags(C -Wno-sometimes-uninitialized)
+	endif()
+	
+	if(OPENMP AND (${CMAKE_C_COMPILER_VERSION} VERSION_LESS 3.7))
+		message(FATAL_ERROR "Clang versions earlier than 3.7 do not support OpenMP!  Disable it or change compilers!")
+	endif()
+		
+endif()
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
+	add_flags(CXX -Wall -Wno-unused-function)
+	
+	list(APPEND OPT_CXXFLAGS "-mtune=native")
+	
+	if(CROSSCOMPILE)
+		set(CMAKE_CXX_COMPILER_TARGET ${TARGET_TRIPLE})
+	endif()
+	
+	if(NOT UNINITIALIZED_WARNINGS)
+		add_flags(CXX -Wno-sometimes-uninitialized)
+	endif()
+	
+	if(TARGET_OSX)
+		# on OS X, Python will link pytraj's extension modules to libc++, so cpptraj needs to use the same standard library
+		add_flags(CXX -stdlib=libc++)
+		if(NOT $ENV{CONDA_BUILD})
+			set(CMAKE_OSX_DEPLOYMENT_TARGET 10.7)
+		endif()
+	endif()	
+	
+	if(OPENMP AND (${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 3.7))
+		message(FATAL_ERROR "Clang versions earlier than 3.7 do not support OpenMP!  Disable it or change compilers!")
+	endif()
+endif()
+
+#msvc
+#---------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+	add_flags(C /D_CRT_SECURE_NO_WARNINGS /MP)
+	
+	set(OPT_CFLAGS "/Ox")
```

Comment:
Is `OPT_CFLAGS` a CMAKE built-in variable?

Reply:
OPT_CFLAGS is a variable containing optimized C flags, it's created by the amber buildscripts.